### PR TITLE
Saml conf

### PR DESCRIPTION
**Pull request by Anuj359**
### Describe your changes :

I worked on the openmetadata charts because saml configuration were to be added in helm.

#
### Frontend Preview (Screenshots) :
image

#
### Type of change :

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: charts/openmetadata/values.yaml [hunk 1]
```diff
@@ -116,6 +116,38 @@ global:
...
+        tokenValidity: 3600
+        sendEncryptedNameId: false
+        sendSignedAuthRequest: false
+        signSpMetadata: false
+        wantMessagesSigned: false
+        wantAssertionsSigned: false
+        wantAssertionEncrypted: false
+        wantNameIdEncrypted: false
+        keyStoreFilePath: ""
+        keyStoreAlias: ""
```
Comment (akash-jain-10) [inline charts/openmetadata/values.yaml:147]: This should also be a secret!

File: charts/openmetadata/values.yaml [hunk 2]
```diff
@@ -116,6 +116,38 @@ global:
...
+          secretKey: ""
+        authorityUrl: "http://localhost:8585/api/v1/saml/login"
+        nameId: "urn:oasis:names:tc:SAML:2.0:nameid-format:emailAddress"
+      sp:
+        entityId: "http://localhost:8585/api/v1/saml/metadata"
+        acs: "http://localhost:8585/api/v1/saml/acs"
+        spX509Certificate: 
+          secretRef: ""
+          secretKey: ""
+        callback: "http://localhost:8585/saml/callback"
```
Comment (akash-jain-10) [inline charts/openmetadata/values.yaml:135]:
```suggestion
        callback: "http://openmetadata.default.svc.cluster.local:8585/saml/callback"
```

Update to kubernetes service fqn urls

File: charts/openmetadata/values.yaml [hunk 3]
```diff
@@ -116,6 +116,38 @@ global:
...
+        entityId: ""
+        ssoLoginUrl: ""
+        idpX509Certificate:
+          secretRef: ""
+          secretKey: ""
+        authorityUrl: "http://localhost:8585/api/v1/saml/login"
+        nameId: "urn:oasis:names:tc:SAML:2.0:nameid-format:emailAddress"
+      sp:
+        entityId: "http://localhost:8585/api/v1/saml/metadata"
+        acs: "http://localhost:8585/api/v1/saml/acs"
```
Comment (akash-jain-10) [inline charts/openmetadata/values.yaml:131]:
```suggestion
        acs: "http://openmetadata.default.svc.cluster.local:8585/api/v1/saml/acs"
```

File: charts/openmetadata/values.yaml [hunk 4]
```diff
@@ -116,6 +116,38 @@ global:
...
+      idp:
+        entityId: ""
+        ssoLoginUrl: ""
+        idpX509Certificate:
+          secretRef: ""
+          secretKey: ""
+        authorityUrl: "http://localhost:8585/api/v1/saml/login"
+        nameId: "urn:oasis:names:tc:SAML:2.0:nameid-format:emailAddress"
+      sp:
+        entityId: "http://localhost:8585/api/v1/saml/metadata"
```
Comment (akash-jain-10) [inline charts/openmetadata/values.yaml:130]:
```suggestion
        entityId: "http://openmetadata.default.svc.cluster.local:8585/api/v1/saml/metadata"
```

File: charts/openmetadata/values.yaml [hunk 5]
```diff
@@ -116,6 +116,38 @@ global:
...
           examineValidityDates: true
+    samlConfiguration:
+      debugMode: false
+      idp:
+        entityId: ""
+        ssoLoginUrl: ""
+        idpX509Certificate:
+          secretRef: ""
+          secretKey: ""
+        authorityUrl: "http://localhost:8585/api/v1/saml/login"
```
Comment (akash-jain-10) [inline charts/openmetadata/values.yaml:127]:
```suggestion
        authorityUrl: "http://openmetadata.default.svc.cluster.local:8585/api/v1/saml/login"
```

File: charts/openmetadata/values.yaml [hunk 6]
```diff
@@ -116,6 +116,38 @@ global:
           verifyHostname: true
         trustAllConfig:
           examineValidityDates: true
+    samlConfiguration:
```
Comment (akash-jain-10) [inline charts/openmetadata/values.yaml:119]:
```suggestion
    saml:
```

File: charts/openmetadata/values.schema.json [hunk 7]
```diff
@@ -233,6 +234,116 @@
                                     }
                                 }
                             }
+                        },
+                        "samlConfiguration": {
```
Comment (akash-jain-10) [inline charts/openmetadata/values.schema.json:238]: Additional Properties should be set to `false` here.

**Review by akash-jain-10 — CHANGES_REQUESTED** [review]
Added few comments around the implementation! Let's fix those and move further here! Thanks.

**Review by mohityadav766 — APPROVED** [review]

**Pull request merged**